src/http/handler/BaseHistoryHttpHandler.java

```diff
@@ -13,13 +12,13 @@ public void handle(HttpExchange exchange) throws IOException {
         String method = exchange.getRequestMethod();
         switch (method) {
             case "GET":
-                getHandle();
+                getHandle(exchange);
                 break;
             default:
                 sendMethodNotAllowed(exchange);
         }
     }
 
-    abstract void getHandle();
+    abstract void getHandle(HttpExchange exchange) throws IOException;
```

Comment:
Можно еще больше избавиться от дублирований если sendMessage вынести сюда в метод handle, так как он повторяется в History и Prioritized хэндлерах. Для этого данный метод должен возвращать json строку вот так:
```suggestion
    protected abstract String getHandle(HttpExchange exchange);
```